src/platform/platform_linux.cpp

```diff
@@ -110,32 +110,50 @@ bool is_ethernet(const QDir& net_dir)
            get_net_devtype(net_dir).isEmpty();
 }
 
-mp::NetworkInterfaceInfo get_network(const QDir& net_dir)
+mp::optional<mp::NetworkInterfaceInfo> get_network(const QDir& net_dir)
 {
     static const auto bridge_fname = QStringLiteral("brif");
+    auto id = net_dir.dirName().toStdString();
 
-    std::string type, description;
-    std::vector<std::string> links;
     if (auto bridge = "bridge"; net_dir.exists(bridge))
     {
-        type = bridge;
+        std::vector<std::string> links;
         QStringList bridge_members = QDir{net_dir.filePath(bridge_fname)}.entryList(QDir::NoDotAndDotDot | QDir::Dirs);
 
         links.reserve(bridge_members.size());
         std::transform(bridge_members.cbegin(), bridge_members.cend(), std::back_inserter(links),
                        [](const QString& interface) { return interface.toStdString(); });
 
-        description = bridge_members.isEmpty() ? "Empty network bridge"
-                                               : fmt::format("Network bridge with {}", bridge_members.join(", "));
+        auto description = bridge_members.isEmpty() ? "Empty network bridge"
+                                                    : fmt::format("Network bridge with {}", bridge_members.join(", "));
+        return {{std::move(id), bridge, std::move(description), std::move(links)}};
     }
     else if (is_ethernet(net_dir))
+        return {{std::move(id), "ethernet", "Ethernet device"}};
+
+    return mp::nullopt;
+}
+
+void update_bridges(std::map<std::string, mp::NetworkInterfaceInfo>& networks)
+{
+    for (auto& item : networks)
     {
-        type = "ethernet";
-        description = "Ethernet device";
+        if (auto& net = item.second; !net.links.empty())
+        {
+            auto& links = net.links;
+            auto is_unknown = [&networks](const std::string& id) {
+                auto same_as = [&id](const auto& other) { return other.first == id; };
+                return std::find_if(networks.cbegin(), networks.cend(), same_as) == networks.cend();
+            };
+            links.erase(std::remove_if(links.begin(), links.end(), is_unknown),
+                        links.end()); // filter links to networks we don't recognize
+
+            assert(net.type == "bridge");
+            net.description =
+                links.empty() ? "Empty network bridge"
```

Comment:
This should probably just say "Network bridge" - it's not necessarily empty:

```plain
$ multipass networks
Name             Type      Description
enx00133b4909cd  ethernet  Ethernet device
lxdbr0           bridge    Empty network bridge
mpbr0            bridge    Network bridge for Multipass
virbr0           bridge    Empty network bridge

$ brctl show
bridge name     bridge id               STP enabled     interfaces
lxdbr0          8000.00163e092819       no              veth3fe7576d
mpbr0           8000.00163eba1a1b       no
virbr0          8000.525400f99bdb       yes             virbr0-nic
```

Reply:
Hmm, right, now that we're filtering things we shouldn't be so specific in that description. I'll update that.

Reply:
Done.